multimodal_knowledge_retriever/tests/multimodal_knowledge_retriever_test.py

```diff
@@ -0,0 +1,135 @@
+    test_conversation_chain_creation: checks if the conversation chain is not None
+    test_question_answering: checks if the answer and source_documents exist and are not empty
+
+Usage:
+    python tests/multimodal_knowledge_retriever_test.py
+
+Returns:
+    0 if all tests pass, or a positive integer representing the number of failed tests.
+"""
+
+import os
+import sys
+import logging
+import time
+import unittest
+from typing import List, Dict, Any
+
+# Setup logging
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(levelname)s - %(message)s')
+logger = logging.getLogger(__name__)
+
+current_dir = os.path.dirname(os.path.abspath(__file__))
+kit_dir = os.path.abspath(os.path.join(current_dir, ".."))
+repo_dir = os.path.abspath(os.path.join(kit_dir, ".."))
+
+sys.path.append(kit_dir)
+sys.path.append(repo_dir)
+
+from multimodal_knowledge_retriever.src.multimodal import MultimodalRetrieval
+
+
+IMAGE_TEST_DATA_PATH = os.path.join(kit_dir, "data/sample_docs")
+PDF_TEST_DATA = os.path.join(IMAGE_TEST_DATA_PATH, "invoicesample.pdf")
+
+class MKRTestCase(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.time_start = time.time()
+        cls.multimodal_retriever = MultimodalRetrieval()
+        cls.text_docs, cls.table_docs, cls.image_paths = cls.parse_data()
+        cls.vectorstore = cls.create_vectorstore()
+        cls.chain = cls.create_retrieval_chain()
+
+    @classmethod
+    def parse_data(cls):
+        raw_pdf_elements, _ = cls.multimodal_retriever.extract_pdf(PDF_TEST_DATA)
+        text_docs, table_docs, image_paths = cls.multimodal_retriever.process_raw_elements(raw_pdf_elements, [IMAGE_TEST_DATA_PATH])
+        logger.info(f'Number of chunks of text: {len(text_docs)}')
+        logger.info(f'Number of chunks of table: {len(table_docs)}')
+        logger.info(f'Number of images: {len(image_paths)}')
+        return text_docs, table_docs, image_paths
+
+    @classmethod
+    def create_vectorstore(cls):
+        retriever = cls.multimodal_retriever.create_vectorstore()
+        return cls.multimodal_retriever.vectorstore_ingest(retriever, cls.text_docs, cls.table_docs, cls.image_paths, summarize_texts=True, summarize_tables=True)
+
+    @classmethod
+    def create_retrieval_chain(cls):
+        return cls.multimodal_retriever.get_retrieval_chain(cls.vectorstore, image_retrieval_type="summary")
+
+    def test_document_parsing(self):
+        self.assertTrue(len(self.text_docs) > 0, f"Parsed {len(self.text_docs)} chunks of text")
+        self.assertTrue(len(self.table_docs) > 0, f"Parsed {len(self.table_docs)} chunks of table text")
+        self.assertTrue(len(self.image_paths) > 0, f"Parsed {len(self.image_paths)} images")
+
+    def test_vector_store_creation(self):
+        self.assertIsNotNone(self.vectorstore, "Vector store created successfully")
+
+    def test_conversation_chain_creation(self):
+        self.assertIsNotNone(self.chain, "Conversation chain created successfully")
```

Comment:
Same thing. Should be:
 self.assertIsNotNone(self.chain, "Conversation chain could not be created")